docs/outline.md

```diff
@@ -0,0 +1,59 @@
+- [0. Overview](#0-overview)
+- [1. Handling Structured Text Data](#1-handling-structured-text-data)
+- [2. Processing Raw Text](#2-processing-raw-text)
+  - [Reader](#reader)
+  - [Tokenizer](#tokenizer)
+  - [Ontology](#ontology)
+  - [Dataset](#dataset)
+- [3. Building a NLP Pipeline](#3-building-a-nlp-pipeline)
+  - [Text Classification](#text-classification)
+- [4. Linguistic Analysis](#4-linguistic-analysis)
+  - [Part-of-Speech Tagging](#part-of-speech-tagging)
+- [5. Language Modeling with External Library](#5-language-modeling-with-external-library)
+- [6. Application](#6-application)
+  - [1. Information Extraction](#1-information-extraction)
+    - [Named Entity Recognition](#named-entity-recognition)
+    - [Relation Extraction](#relation-extraction)
+    - [Indexing](#indexing)
+    - [Ranking](#ranking)
+  - [2. Machine Translation](#2-machine-translation)
+  - [3. Text Generation](#3-text-generation)
+  - [4. Question Answering System](#4-question-answering-system)
+  - [5. Chatbot](#5-chatbot)
+- [7. More data format](#7-more-data-format)
+  - [Audio Data](#audio-data)
+  - [Image Data](#image-data)
```

Comment:
You probably don't need a TOC since sphinx will have a sidebar. but the TOC is good

Reply:
ok I think we can leave it there and delete it later if it's not useful.